### PR TITLE
GC: Add test to validate imported data isn't deleted

### LINKED ISSUE #4757: GC: add tests to validate that imported data is not GCed
Our code covers the case of imported data, and GC is designed not to collect it. However we would like to add a test that enforces that.

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
### Linked Issue

Closes #4757 

---

## Change Description

Add a test to validate that imported data isn't swept.
Also added a piece of code to remove a `/` prefix from an object if the storage namespace **doesn't** end with `/`.

### CODE REVIEW

**Review by johnnyaug — APPROVED** [review]
🚀

File: clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala [hunk 1]
```diff
@@ -145,6 +145,51 @@ class GarbageCollectorSpec extends AnyFunSpec with Matchers with SparkSessionSet
...
+          import spark.implicits._
+          val storageNamespace = "s3://repo-ns"
+          val importedDataS3Path = "s3://bucket/of/imported/data"
+          val repoName = "repo-name"
+          val rangeGetter = new ARangeGetter(repoName,
+                                             null,
+                                             Map(
+                                               "aaa" -> Seq("a1",
+                                                            s"$importedDataS3Path/a2",
+                                                            s"$storageNamespace/a3"
```
Comment (arielshaqed) [inline clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala:161]: If you define these as functions, you might instead write
```suggestion
                                                            importedDataS3Path("a2"),
                                                            storageNamespace("a3")
```
Might be clearer?

File: clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala [hunk 2]
```diff
@@ -145,6 +145,51 @@ class GarbageCollectorSpec extends AnyFunSpec with Matchers with SparkSessionSet
...
+                                                            "c2"
+                                                           )
+                                             )
+                                            )
+          val gc = new GarbageCollector(rangeGetter)
+
+          val actualToDelete = gc.getAddressesToDelete(Seq("aaa", "222", "bbb").toDS,
+                                                       Seq[String]().toDS,
+                                                       repoName,
+            s"$storageNamespace/",
```
Comment (arielshaqed) [inline clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala:183]: Nit: Strange indentation.  Tabs maybe?

**Review by arielshaqed — APPROVED** [review]
Thanks!  Much needed test coverage, I feel safer already..

**Comment by arielshaqed** [comment]
> Also added a piece of code to remove a `/` prefix from an object if the storage namespace **doesn't** end with `/`.

Not sure, did you end up including this?!

**Pull request merged**